File: src/main/java/cn/dsnbo/bedrockplayersupport/form/CMIForm.java
package cn.dsnbo.bedrockplayersupport.form;

import cn.dsnbo.bedrockplayersupport.BedrockPlayerSupport;
import com.Zrips.CMI.CMI;
import com.Zrips.CMI.Containers.CMIUser;
import com.Zrips.CMI.Modules.Homes.CmiHome;
import org.bukkit.entity.Player;
import org.geysermc.cumulus.form.SimpleForm;

import java.util.LinkedHashMap;

public class CMIForm {

    public static void sendHomeForm(Player player) {
        CMIUser cmiUser = CMI.getInstance().getPlayerManager().getUser(player);
        LinkedHashMap<String, CmiHome> playerHomesList = cmiUser.getHomes();
        SimpleForm.Builder form = SimpleForm.builder()
                .title("§6§l我的家")
                .validResultHandler(simpleFormResponse -> {
                    player.chat("/home " + simpleFormResponse.clickedButton().text());
                });
        for (String homeName : playerHomesList.keySet()) {
            form.button(homeName);
        }
        BedrockPlayerSupport.getFloodgateApi().sendForm(player.getUniqueId(), form);
    }

}

File: src/main/kotlin/cc/dsnb/bedrockplayersupport/form/basic/CMIForm.kt
package cc.dsnb.bedrockplayersupport.form.basic

import cc.dsnb.bedrockplayersupport.BedrockPlayerSupport
import cc.dsnb.bedrockplayersupport.util.StringUtil
import com.Zrips.CMI.CMI
import org.bukkit.entity.Player
import org.geysermc.cumulus.form.SimpleForm

/**
 * @author DongShaoNB
 */
class CMIForm {

    fun sendDelHomeForm(player: Player) {
        val cmiUser = CMI.getInstance().playerManager.getUser(player)
        val playerHomesList = cmiUser.homes
        val form = SimpleForm.builder()
            .title(
                StringUtil.formatTextToString(
                    player,
                    BedrockPlayerSupport.langConfigManager.getConfigData().delHomeFormTitle()
                )
            )
            .validResultHandler { simpleFormResponse ->
                player.chat("/removehome ${simpleFormResponse.clickedButton().text()}")
            }
        playerHomesList.forEach { form.button(it.key) }
        BedrockPlayerSupport.floodgateApi.sendForm(player.uniqueId, form)
    }

    fun sendGoHomeForm(player: Player) {
        val cmiUser = CMI.getInstance().playerManager.getUser(player)
        val playerHomesList = cmiUser.homes
        val form = SimpleForm.builder()
            .title(
                StringUtil.formatTextToString(
                    player,
                    BedrockPlayerSupport.langConfigManager.getConfigData().goHomeFormTitle()
                )
            )
            .validResultHandler { simpleFormResponse ->
                player.chat("/home ${simpleFormResponse.clickedButton().text()}")
            }
        playerHomesList.forEach { form.button(it.key) }
        BedrockPlayerSupport.floodgateApi.sendForm(player.uniqueId, form)
    }

    fun sendWarpForm(player: Player) {
        val warps = CMI.getInstance().warpManager.warps
        val form = SimpleForm.builder()
            .title(
                StringUtil.formatTextToString(
                    player,
                    BedrockPlayerSupport.langConfigManager.getConfigData().warpFormTitle()
                )
            )
            .validResultHandler { simpleFormResponse ->
                player.chat("/warp ${simpleFormResponse.clickedButton().text()}")
            }
        warps.forEach { form.button(it.key) }
        BedrockPlayerSupport.floodgateApi.sendForm(player.uniqueId, form)
    }

    fun sendKitForm(player: Player) {
        val kits = CMI.getInstance().kitsManager.kitMap
        val form = SimpleForm.builder()
            .title(
                StringUtil.formatTextToString(
                    player,
                    BedrockPlayerSupport.langConfigManager.getConfigData().kitFormTitle()
                )
            )
            .validResultHandler { simpleFormResponse ->
                player.chat("/kit ${simpleFormResponse.clickedButton().text()}")
            }
        kits.forEach { form.button(it.key) }
        BedrockPlayerSupport.floodgateApi.sendForm(player.uniqueId, form)
    }

}